onnxruntime/core/optimizer/transpose_optimizer/optimizer_utils.h

```diff
@@ -88,7 +88,7 @@ const std::unordered_set<std::string_view>& GetORTLayoutSensitiveOps();
 /// This is called after layout transformation if new nodes are inserted, and again after those are optimized.
 /// </param>
 Status TransformLayoutForEP(Graph& graph, bool& modified, const IExecutionProvider& execution_provider,
-                            const DebugGraphFn& debug_graph_fn = {});
+                            AllocatorPtr allocator, const DebugGraphFn& debug_graph_fn = {});
```

Comment:
>allocator [](http://example.com/codeflow?start=41&length=9)

there are many places we just need a CPU allocator during graph transformation, mainly used to create some temporary buffers.  it looks kind of weird that pass in allocator ptr so many stacks but just for temp buffer creation.

should we consider any more convenience API for it?